pkg/ddevapp/config.go

```diff
@@ -1114,10 +1114,41 @@ func WriteBuildDockerfile(app *DdevApp, fullpath string, userDockerfilePath stri
 		return err
 	}
 
-	// Normal starting content is the arg and base image
-	contents := `
-#ddev-generated - Do not modify this file; your modifications will be overwritten.
+	contents := "#ddev-generated - Do not modify this file; your modifications will be overwritten.\n"
+
+	// Build custom PHP extensions if needed
+	// See /usr/local/bin/build_php_extension.sh
+	phpExtensions := []map[string]string{
+		// compile specific Xdebug version https://github.com/ddev/ddev/issues/6159
+		{
+			"php":     nodeps.PHP80,
+			"name":    "xdebug",
+			"version": "3.2.2",
+			"file":    "/usr/lib/php/20200930/xdebug.so",
+		},
+	}
+	if strings.Contains(fullpath, "webimageBuild") {
+		hasMultiStageBuild := false
+		for _, ext := range phpExtensions {
+			if app.PHPVersion == ext["php"] {
+				if !hasMultiStageBuild {
+					contents = contents + `
+### DDEV-injected custom build for PHP extensions
+ARG BASE_IMAGE="scratch"
+FROM $BASE_IMAGE AS ddev-php-extension-build
+SHELL ["/bin/bash", "-c"]
+`
+					hasMultiStageBuild = true
+				}
+				contents = contents + fmt.Sprintf(`
+RUN /usr/local/bin/build_php_extension.sh "php%s" "%s" "%s" "%s" || true
```

Comment:
I used `|| true` here, so it won't break the whole build for the user.
(It will restore the original extension library file on fail, so `COPY --from` below should never fail.)